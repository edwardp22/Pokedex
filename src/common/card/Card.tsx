import React from "react";

import {
  Card as CardMUI,
  CardContent,
  CardMedia,
  Grid,
  Typography,
} from "@mui/material";

import { Star } from "../";
import { spritesBaseURL } from "../../shared";
import { TPokemon } from "../../types";
import { IPokemonSprites } from "../../types/pokemonSprites";
import styles from "./card.module.css";

export default function Card({
  pokemon,
  onClick,
}: {
  pokemon: TPokemon;
  onClick: (pokemon: TPokemon) => void;
}) {
  const { star, title, imageStyle } = styles;
  const { name, pokemon_v2_pokemonsprites } = pokemon;
  const sprites: IPokemonSprites = pokemon_v2_pokemonsprites.length
    ? JSON.parse(pokemon_v2_pokemonsprites[0].sprites)
    : undefined;
  const oficialArtwork = sprites.other?.["official-artwork"].front_default;
  const imageURL = spritesBaseURL + oficialArtwork?.replaceAll("/media/", "");

  return (
    <Grid item sm={3}>
      <CardMUI>
        <CardContent>
          <div className={star}>
            <Star onClick={() => onClick(pokemon)} />
          </div>

          <Typography variant="h6" className={title}>
            {name}
          </Typography>

          <CardMedia className={imageStyle} image={imageURL} />
        </CardContent>
      </CardMUI>
    </Grid>
  );
}
