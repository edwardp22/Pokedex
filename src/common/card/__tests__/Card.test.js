import renderer from "react-test-renderer";
import Card from "../Card";

const newPokemon = {
  id: 1,
  name: "bulbasaur",
  pokemon_v2_pokemonsprites: [
    {
      sprites:
        '{"front_default": "/media/sprites/pokemon/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/shiny/1.png", "front_shiny_female": null, "back_default": "/media/sprites/pokemon/back/1.png", "back_female": null, "back_shiny": "/media/sprites/pokemon/back/shiny/1.png", "back_shiny_female": null, "other": {"dream_world": {"front_default": "/media/sprites/pokemon/other/dream-world/1.svg", "front_female": null}, "home": {"front_default": "/media/sprites/pokemon/other/home/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/other/home/shiny/1.png", "front_shiny_female": null}, "official-artwork": {"front_default": "/media/sprites/pokemon/other/official-artwork/1.png", "front_shiny": "/media/sprites/pokemon/other/official-artwork/shiny/1.png"}}, "versions": {"generation-i": {"red-blue": {"front_default": "/media/sprites/pokemon/versions/generation-i/red-blue/1.png", "front_gray": "/media/sprites/pokemon/versions/generation-i/red-blue/gray/1.png", "back_default": "/media/sprites/pokemon/versions/generation-i/red-blue/back/1.png", "back_gray": "/media/sprites/pokemon/versions/generation-i/red-blue/back/gray/1.png", "front_transparent": "/media/sprites/pokemon/versions/generation-i/red-blue/transparent/1.png", "back_transparent": "/media/sprites/pokemon/versions/generation-i/red-blue/transparent/back/1.png"}, "yellow": {"front_default": "/media/sprites/pokemon/versions/generation-i/yellow/1.png", "front_gray": "/media/sprites/pokemon/versions/generation-i/yellow/gray/1.png", "back_default": "/media/sprites/pokemon/versions/generation-i/yellow/back/1.png", "back_gray": "/media/sprites/pokemon/versions/generation-i/yellow/back/gray/1.png", "front_transparent": "/media/sprites/pokemon/versions/generation-i/yellow/transparent/1.png", "back_transparent": "/media/sprites/pokemon/versions/generation-i/yellow/transparent/back/1.png"}}, "generation-ii": {"crystal": {"front_default": "/media/sprites/pokemon/versions/generation-ii/crystal/1.png", "front_shiny": "/media/sprites/pokemon/versions/generation-ii/crystal/shiny/1.png", "back_default": "/media/sprites/pokemon/versions/generation-ii/crystal/back/1.png", "back_shiny": "/media/sprites/pokemon/versions/generation-ii/crystal/back/shiny/1.png", "front_transparent": "/media/sprites/pokemon/versions/generation-ii/crystal/transparent/1.png", "front_shiny_transparent": "/media/sprites/pokemon/versions/generation-ii/crystal/transparent/shiny/1.png", "back_transparent": "/media/sprites/pokemon/versions/generation-ii/crystal/transparent/back/1.png", "back_shiny_transparent": "/media/sprites/pokemon/versions/generation-ii/crystal/transparent/back/shiny/1.png"}, "gold": {"front_default": "/media/sprites/pokemon/versions/generation-ii/gold/1.png", "front_shiny": "/media/sprites/pokemon/versions/generation-ii/gold/shiny/1.png", "back_default": "/media/sprites/pokemon/versions/generation-ii/gold/back/1.png", "back_shiny": "/media/sprites/pokemon/versions/generation-ii/gold/back/shiny/1.png", "front_transparent": "/media/sprites/pokemon/versions/generation-ii/gold/transparent/1.png"}, "silver": {"front_default": "/media/sprites/pokemon/versions/generation-ii/silver/1.png", "front_shiny": "/media/sprites/pokemon/versions/generation-ii/silver/shiny/1.png", "back_default": "/media/sprites/pokemon/versions/generation-ii/silver/back/1.png", "back_shiny": "/media/sprites/pokemon/versions/generation-ii/silver/back/shiny/1.png", "front_transparent": "/media/sprites/pokemon/versions/generation-ii/silver/transparent/1.png"}}, "generation-iii": {"emerald": {"front_default": "/media/sprites/pokemon/versions/generation-iii/emerald/1.png", "front_shiny": "/media/sprites/pokemon/versions/generation-iii/emerald/shiny/1.png"}, "firered-leafgreen": {"front_default": "/media/sprites/pokemon/versions/generation-iii/firered-leafgreen/1.png", "front_shiny": "/media/sprites/pokemon/versions/generation-iii/firered-leafgreen/shiny/1.png", "back_default": "/media/sprites/pokemon/versions/generation-iii/firered-leafgreen/back/1.png", "back_shiny": "/media/sprites/pokemon/versions/generation-iii/firered-leafgreen/back/shiny/1.png"}, "ruby-sapphire": {"front_default": "/media/sprites/pokemon/versions/generation-iii/ruby-sapphire/1.png", "front_shiny": "/media/sprites/pokemon/versions/generation-iii/ruby-sapphire/shiny/1.png", "back_default": "/media/sprites/pokemon/versions/generation-iii/ruby-sapphire/back/1.png", "back_shiny": "/media/sprites/pokemon/versions/generation-iii/ruby-sapphire/back/shiny/1.png"}}, "generation-iv": {"diamond-pearl": {"front_default": "/media/sprites/pokemon/versions/generation-iv/diamond-pearl/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-iv/diamond-pearl/shiny/1.png", "front_shiny_female": null, "back_default": "/media/sprites/pokemon/versions/generation-iv/diamond-pearl/back/1.png", "back_female": null, "back_shiny": "/media/sprites/pokemon/versions/generation-iv/diamond-pearl/back/shiny/1.png", "back_shiny_female": null}, "heartgold-soulsilver": {"front_default": "/media/sprites/pokemon/versions/generation-iv/heartgold-soulsilver/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-iv/heartgold-soulsilver/shiny/1.png", "front_shiny_female": null, "back_default": "/media/sprites/pokemon/versions/generation-iv/heartgold-soulsilver/back/1.png", "back_female": null, "back_shiny": "/media/sprites/pokemon/versions/generation-iv/heartgold-soulsilver/back/shiny/1.png", "back_shiny_female": null}, "platinum": {"front_default": "/media/sprites/pokemon/versions/generation-iv/platinum/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-iv/platinum/shiny/1.png", "front_shiny_female": null, "back_default": "/media/sprites/pokemon/versions/generation-iv/platinum/back/1.png", "back_female": null, "back_shiny": "/media/sprites/pokemon/versions/generation-iv/platinum/back/shiny/1.png", "back_shiny_female": null}}, "generation-v": {"black-white": {"front_default": "/media/sprites/pokemon/versions/generation-v/black-white/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-v/black-white/shiny/1.png", "front_shiny_female": null, "back_default": "/media/sprites/pokemon/versions/generation-v/black-white/back/1.png", "back_female": null, "back_shiny": "/media/sprites/pokemon/versions/generation-v/black-white/back/shiny/1.png", "back_shiny_female": null, "animated": {"front_default": "/media/sprites/pokemon/versions/generation-v/black-white/animated/1.gif", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-v/black-white/animated/shiny/1.gif", "front_shiny_female": null, "back_default": "/media/sprites/pokemon/versions/generation-v/black-white/animated/back/1.gif", "back_female": null, "back_shiny": "/media/sprites/pokemon/versions/generation-v/black-white/animated/back/shiny/1.gif", "back_shiny_female": null}}}, "generation-vi": {"omegaruby-alphasapphire": {"front_default": "/media/sprites/pokemon/versions/generation-vi/omegaruby-alphasapphire/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-vi/omegaruby-alphasapphire/shiny/1.png", "front_shiny_female": null}, "x-y": {"front_default": "/media/sprites/pokemon/versions/generation-vi/x-y/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-vi/x-y/shiny/1.png", "front_shiny_female": null}}, "generation-vii": {"ultra-sun-ultra-moon": {"front_default": "/media/sprites/pokemon/versions/generation-vii/ultra-sun-ultra-moon/1.png", "front_female": null, "front_shiny": "/media/sprites/pokemon/versions/generation-vii/ultra-sun-ultra-moon/shiny/1.png", "front_shiny_female": null}, "icons": {"front_default": "/media/sprites/pokemon/versions/generation-vii/icons/1.png", "front_female": null}}, "generation-viii": {"icons": {"front_default": "/media/sprites/pokemon/versions/generation-viii/icons/1.png", "front_female": null}}}}',
    },
  ],
  pokemon_v2_pokemontypes: [
    {
      pokemon_v2_type: {
        name: "grass",
      },
    },
    {
      pokemon_v2_type: {
        name: "poison",
      },
    },
  ],
};

it("should render the pokemon in the card", () => {
  const component = renderer.create(<Card pokemon={newPokemon} />);
  expect(component.toJSON()).toMatchSnapshot();
});
