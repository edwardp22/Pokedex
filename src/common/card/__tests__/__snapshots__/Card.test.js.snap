// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the pokemon in the card 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3 css-13i4rnv-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        className="star"
      >
        <i
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium starUnselected css-i4bv87-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </i>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        bulbasaur
      </h6>
      <div
        className="MuiCardMedia-root imageStyle css-16cvb1l-MuiCardMedia-root"
        role="img"
        style={
          Object {
            "backgroundImage": "url(\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png\\")",
          }
        }
      />
    </div>
  </div>
</div>
`;
